src/CMakeLists.txt

```diff
@@ -434,32 +435,31 @@ if(WITH_JAEGER)
 		    ${CMAKE_BINARY_DIR}/external/lib/libyaml-cpp.so.0.6.2)
     execute_process(COMMAND bash -c "grep -q 'yaml-cpp' debian/libjaeger.install || echo 'usr/lib/libyaml-cpp.so.*' >> debian/libjaeger.install"
 		    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
-    add_library(yaml-cpp::yaml-cpp SHARED IMPORTED)
-    add_dependencies(yaml-cpp::yaml-cpp yaml-cpp)
     set_library_properties_for_external_project(yaml-cpp::yaml-cpp yaml-cpp)
     list(APPEND dependencies "yaml-cpp")
+    set(jaeger_base yaml-cpp::yaml-cpp)
   endif()
-  include(IncludeJaeger)
-  add_library(jaeger-base INTERFACE)
-  add_dependencies(common-objs
-  yaml-cpp::yaml-cpp
-  opentracing::libopentracing
-  thrift::libthrift
-  jaegertracing::libjaegertracing)
-  target_link_libraries(jaeger-base INTERFACE
-  yaml-cpp::yaml-cpp
+
+  include(BuildJaeger)
+  set(jaeger_base ${jaeger_base}
   opentracing::libopentracing
-  thrift::libthrift
   jaegertracing::libjaegertracing)
-  include_directories(SYSTEM ${CMAKE_BINARY_DIR}/external/include)
-  #with CMake 3.12+ the following can be replaced by:
+  message(" jaeger build targets: " ${jaeger_base})
+  #with CMake 3.12+ the above can be replaced by:
   #target_link_libraries(common-objs jaeger-base)
-   list(APPEND jaeger_libs
+  # for testing jaeger builds
+  add_custom_target(jaeger_base DEPENDS ${jaeger_base})
+
+  add_library(jaeger-base INTERFACE)
+  target_link_libraries(jaeger-base INTERFACE ${jaeger_base})
+  include_directories(SYSTEM ${CMAKE_BINARY_DIR}/external/include)
+  target_link_libraries(common-objs jaeger-base)
```

Comment:
again, please break this change into smaller pieces.

Reply:
done